content/en/users/cloud-compute/openstack/_index.md

```diff
@@ -215,107 +215,64 @@ openstack server create --flavor <flavor> \
 
 ## Terraform
 
-[Terraform](https://terraform.io) supports EGI Cloud OpenStack providers by
-using valid access tokens for Keystone. For using this, just configure your
-provider as usual in Terraform, but do not include user/password information:
+[Terraform](https://registry.terraform.io/providers/terraform-provider-openstack/openstack/)
+supports EGI Cloud OpenStack providers by using valid access tokens for Keystone.
+For using this, just configure your provider as usual in Terraform, but do not
+include user/password information. Instead, use the [fedcloud](../../getting-started/cli/)
+client to configure environment variables as follows:
 
-```terraform
-# Configure the OpenStack Provider
-provider "openstack" {
-  project_id = "<your project id>"
-  auth_url    = "http://<your keystone url>/v3"
-}
+```shell
+# export OS_AUTH_URL and OS_PROJECT_ID with
+$ eval "$(fedcloud site show-project-id --site <NAME_OF_SITE> --vo <NAME_OF_VO>)"
 
-# Create a server
-resource "openstack_compute_instance_v2" "test-server" {
-  # ...
-}
+# now get a valid token
+$ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
+                    token issue -c id -f value)
 ```
 
-when launching Terraform, set the `OS_TOKEN` environment variable to a valid
-token as shown in :ref:OpenStack token for other clients. You may also set the
-Keystone URL and project ID in the `OS_AUTH_URL` and `OS_PROJECT_ID` environment
-variables:
+Here is a sample `main.tf` configuration file for Terraform:
 
 ```terraform
-provider "openstack" {
-}
-
-data "openstack_images_image_v2" "ubuntu16" {
-  most_recent = true
-
-  properties {
-    APPLIANCE_MPURI = "https://appdb.egi.eu/store/vo/image/8df7ba00-8467-57aa-bf1e-05754a2a73bf:6428/"
+terraform {
+  required_version = "1.1.2"
```

Comment:
Is it worth to pin the versions here? Just thinking on how to make this guide last